cypress/e2e/timeline/timeline.spec.ts

```diff
@@ -302,6 +302,96 @@ describe("Timeline", () => {
+                "created and configured the room.",
+            ).should("exist");
+
+            // Send unordered & ordered lists with paragraphs
+            cy.sendEvent(roomId, null, "m.room.message" as EventType, {
+                msgtype: "m.text" as MsgType,
+                body:
+                    "Paragraph above lists\n\n" +
+                    "- item 1\n- item 2\n- item 3\n\n" +
+                    "Paragraph between lists\n\n" +
+                    "1. item 1\n2. item 2\n3. item 3\n\n" +
+                    "Paragraph below lists",
+                format: "org.matrix.custom.html",
+                formatted_body:
+                    "<p>Paragraph above lists</p>\n" +
+                    "<ul>\n<li>item 1</li>\n<li>item 2</li>\n<li>item 3</li>\n</ul>\n" +
+                    "<p>Paragraph between lists</p>\n" +
+                    "<ol>\n<li>item 1</li>\n<li>item 2</li>\n<li>item 3</li>\n</ol>\n" +
+                    "<p>Paragraph below lists</p>\n",
+            });
+
+            // Confirm the paragraph below the lists was rendered
+            cy.get(".mx_EventTile[data-layout=group] .markdown-body p:last-of-type").should(
+                "have.text",
+                "Paragraph below lists",
+            );
+
+            // This is to check block margin of unordered & ordered lists
+            const checkMarginList = () => {
+                cy.get("ul").should("have.css", "margin-block", "0px");
+                cy.get("ol").should("have.css", "margin-block", "0px");
+            };
+
+            // This is to check block margin of paragraphs around the lists except on compact modern layout
+            const checkMarginParagraphsWide = () => {
+                cy.get("p:first-of-type").should("have.css", "margin-block", "0px 16px"); // Paragraph above lists
+                cy.get("p:nth-of-type(2)").should("have.css", "margin-block", "16px"); // Paragraph between lists
+                cy.get("p:last-of-type").should("have.css", "margin-block", "16px 0px"); // Paragraph below lists
+            };
+
+            // Check block margin values of paragraphs and lists on modern layout
+            cy.get(".mx_EventTile[data-layout=group] .markdown-body").within(() => {
+                checkMarginList();
+                checkMarginParagraphsWide();
+            });
+
+            // Exclude timestamp and read marker from snapshot
+            const percyCSS = ".mx_MessageTimestamp, .mx_RoomView_myReadMarker { visibility: hidden !important; }";
+            cy.get(".mx_MainSplit").percySnapshotElement("Lists with paragraphs on modern layout", { percyCSS });
+
+            // Check block margin values of paragraphs and lists on compact modern layout
+            cy.setSettingValue("useCompactLayout", null, SettingLevel.DEVICE, true);
+            cy.get(".mx_EventTile[data-layout=group] .markdown-body").within(() => {
+                cy.get("ul").should("have.css", "margin-block", "0px 4px");
```

Comment:
If ul and ol margin is in fact not expected here, this needs to be replaced as well. https://github.com/matrix-org/matrix-react-sdk/pull/10305/files#r1127766737